src/Conversion/ONNXToKrnl/Math/Softmax.cpp

```diff
@@ -23,14 +23,19 @@ namespace onnx_mlir {
 
 static void emitInnerLoops(KrnlBuilder &createKrnl, int64_t numberOfLoops,
     SmallVectorImpl<IndexExpr> &Lbs, SmallVectorImpl<IndexExpr> &Ubs,
-    ValueRange outerIndices, Value input, Value alloc, Value sumOp, Value maxOp,
-    int64_t axis, bool coerced = true) {
+    ValueRange outerIndices, Value input, Value alloc, Value zero,
+    Value negInfinity, int64_t axis, bool coerced = true) {
   int64_t rank = alloc.getType().cast<MemRefType>().getRank();
 
+  ValueRange maxInits = ValueRange(negInfinity);
   // Compute the maximum value along axis.
   ValueRange maxLoops = createKrnl.defineLoops(numberOfLoops);
-  createKrnl.iterateIE(maxLoops, maxLoops, Lbs, Ubs,
+  auto maxLoop = createKrnl.iterateIE(maxLoops, maxLoops, Lbs, Ubs, maxInits,
       [&](KrnlBuilder &createKrnl, ValueRange maxIndices) {
+        Block *iterEntryBB = createKrnl.getBuilder().getBlock();
+        // Get last argument for the iterate body.
+        Value iterArg = iterEntryBB->getArguments().back();
```

Comment:
I believe we would access the iteration arguments more frequently. How about creating a krnl op to query an iterArg, say `KrnlGetIterationArgs`, or we can put it into an argument of `createKrnl.iterateIE` similar to `maxIndices` and we can use it inside the body without explicitly getting it.

Reply:
Updated as argument like maxIndices for the bodybuilder of KrnlIterateOp to access the iteration arguments.
KrnlIterateOp::getRegionIterArgs was already added to access the iteration arguments of KrnlIterateOp.

Reply:
Thank you!